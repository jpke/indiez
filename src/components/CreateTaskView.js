import React from 'react';

//view called by CreateTaskContainer
const CreateTaskView = (props) => (
  <div id="newTaskForm">
    <h2>{props.newTask._id ? "Edit Task" : "Create New Task"}</h2>
    <form onSubmit={(e) => {
        e.preventDefault();
        props.createTask()
        props.cancelCreateTask()}}>
      <label>Task Name</label>
      <input type="text" name="name" defaultValue={props.newTask.name} onChange={(e) => {props.updateNewTask(e.target.name, e.target.value)}} />
      <label>End Date</label>
      <input type="date" name="endDate" value={props.newTask.end} onChange={(e) => {props.updateNewTask(e.target.name, e.target.value)}} />
      <label>Description</label>
      <textarea type="text" name="description" defaultValue={props.newTask.description} onChange={(e) => {props.updateNewTask(e.target.name, e.target.value)}} />
      <label>Created By</label>
      <input type="text" name="createdBy" defaultValue={props.newTask.createdBy} onChange={(e) => {props.updateNewTask(e.target.name, e.target.value)}} />
      <section className="taskButtons">
        <button onClick={() => {props.cancelCreateTask()}}>Cancel</button>
        <button type="submit">{props.newTask._id ? "Update" : "Create"}</button>
      </section>
    </form>
  </div>
);

export default CreateTaskView;
