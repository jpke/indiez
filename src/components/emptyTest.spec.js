// Must have at least one test file in this directory or Mocha will throw an error.

import React from 'react';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import CreateTaskView from './CreateTaskView';

it('renders correctly', () => {
  const tree = renderer.create(
    <CreateTaskView
      updateNewTask={sinon.spy()}
      createTask={sinon.spy()}
      newTask="New Task"
      cancelCreateTask={sinon.spy()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
