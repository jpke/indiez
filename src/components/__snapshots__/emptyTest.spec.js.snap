exports[`test renders correctly 1`] = `
<div
  id="newTaskForm">
  <h2>
    Create New Task
  </h2>
  <form
    onSubmit={[Function]}>
    <label>
      Task Name
    </label>
    <input
      defaultValue={undefined}
      name="name"
      onChange={[Function]}
      type="text" />
    <label>
      End Date
    </label>
    <input
      defaultValue={undefined}
      name="endDate"
      onChange={[Function]}
      type="date" />
    <label>
      Description
    </label>
    <textarea
      defaultValue={undefined}
      name="description"
      onChange={[Function]}
      type="text" />
    <section
      className="taskButtons">
      <button
        onClick={[Function]}>
        Cancel
      </button>
      <button
        type="submit">
        Create
      </button>
    </section>
  </form>
</div>
`;
